### DISCUSSION TITLE
Should pointer be allowed in carbon?

### DISCUSSION THREAD

**Discussion by martcpp** [carbon-language/carbon-lang#1756] [Polls]
Pointer

**Comment by pmq24** [comment]
Pointers in C++ is one of the most powerful tools that give you complete control of your programs. Being the C++ successor, pointers are a must in Carbon.

**Reply by leikang123** [reply]
The pointer of c++ is very flexible, and it is inherited from c. The system-level language must have pointers, which are close to the hardware. It is better to inherit c++

**Comment by bayrameker** [comment]
adding pointers brings Carbon one step closer to being a C++ successor

**Comment by nacaclanga** [comment]
Pointers are usefull in some capacities. Even Rust has raw pointers besides references for exactly this purpose. They should take a prominent position though.
